client/src/components/client-area/orders/OrdersListItem.vue

```diff
@@ -0,0 +1,269 @@
+            :class="orderStep"
+          >
+            {{ $msTranslate(getOrderStepTranslations(orderStep!).tag) }}
+          </span>
+        </ion-text>
+        <ion-text class="order-header__date subtitles-normal">
+          <span class="date-title body-sm">
+            {{ $msTranslate('clientArea.orders.new.date') }}
+          </span>
+          {{ $msTranslate(I18n.formatDate(request.orderDate, 'short')) }}
+        </ion-text>
+      </div>
+      <!-- order details -->
+      <div
+        v-if="open"
+        class="order-content"
+      >
+        <template v-if="orderStep && orderStep !== OrderStep.Unknown">
+          <order-details
+            :request="request"
+            :order-step="orderStep"
+            :organization="organization"
+          />
+          <order-tracking-progress
+            class="order-content-tracking"
+            :order-step="orderStep"
+          />
+        </template>
+        <template v-else>
+          <ion-text class="error-message">
+            {{ $msTranslate(getOrderStepTranslations(OrderStep.Unknown).description) }}
+          </ion-text>
+        </template>
+      </div>
+    </template>
+
+    <template v-if="error">
+      <ion-text class="error-message">
+        {{ $msTranslate(error) }}
+      </ion-text>
+    </template>
+
+    <template v-if="querying">
+      <div class="skeleton-list">
+        <ion-skeleton-text
+          :animated="true"
+          class="skeleton-item"
+        />
+        <ion-skeleton-text
+          :animated="true"
+          class="skeleton-item"
+        />
+      </div>
+    </template>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { IonText, IonSkeletonText } from '@ionic/vue';
+import { BmsOrganization, BmsAccessInstance, CustomOrderRequest, DataType } from '@/services/bms';
+import { ref, onMounted } from 'vue';
+import OrderDetails from '@/components/client-area/orders/OrderDetails.vue';
+import OrderTrackingProgress from '@/components/client-area/orders/OrderTrackingProgress.vue';
+import { I18n } from 'megashark-lib';
+import { getOrderStep, getOrderStepTranslations, OrderStep } from '@/components/client-area/orders/utils';
+
+const error = ref<string>('');
+const querying = ref(true);
+const orderStep = ref<OrderStep | undefined>();
+const organization = ref<BmsOrganization | undefined>(undefined);
+const open = ref(false);
+
+const props = defineProps<{
+  request: CustomOrderRequest;
+}>();
+
+onMounted(async () => {
+  try {
+    querying.value = true;
+
+    // Set the order step based on the custom order request status
+    orderStep.value = getOrderStep(props.request.status);
+
+    // Here we aim to fetch the organization details based on the request's organizationId.
+    // If the organizationId is not present, we skip the fetching.
+    if (!props.request.organizationId) {
+      return;
+    }
+
+    // Fetch the organization details
+    const orgResp = await BmsAccessInstance.get().listOrganizations();
```

Comment:
Can open an issue to refactorize this. The list of organizations is known by the main page, it shouldn't change that much, so we can avoid a lot of useless querying by just passing the list as a prop.